pywren/wrenhandler.py

```diff
@@ -292,13 +292,18 @@ def consume_stdout(stdout, queue):
         t.start()
 
         stdout = b""
-        while t.isAlive():
+        while t.isAlive() or process.returncode is None:
+            logger.info("Running {} {}".format(time.time(), process.returncode))
             try:
                 line = q.get_nowait()
                 stdout += line
                 logger.info(line)
             except Empty:
                 time.sleep(PROCESS_STDOUT_SLEEP_SECS)
+            process.poll() # this updates retcode but does not block
+            if not t.isAlive() and process.returncode is None:
```

Comment:
I'm not sure I follow this -- if the stdout reader thread is not alive then how can we get output from it ? will the `q.get_nowait()` ever give anything useful in that case ?

Ideally I'd think we want `t.isAlive` and `process.retcode is None` in line 295  ?

Reply:
For this comment, `q.get_nowait()` should fire `except Empty`, is that correct? @Vaishaal